crates/dslab-mp/src/mc/pending_events.rs

```diff
@@ -69,16 +70,29 @@ impl PendingEvents {
 
     /// Returns currently available events, i.e. not blocked by other events (see DependencyResolver).
     pub fn available_events(&self) -> &BTreeSet<McEventId> {
-        &self.available_events
+        if self.directives.is_empty() {
+            &self.available_events
+        } else {
+            &self.directives
```

Comment:
А еще, кстати, можно уменьшить пространство перебора :) опять убираем зависимости
```suggestion
            let result = BTreeSet::default();
            result.insert(self.directives.iter().next().unwrap());
            &result
```

Reply:
Или даже весь блок заменить
```rust
if Some(directive) = self.directives.next() {
    &BTreeSet::from_iter(vec![directive])
} else {
    self.available_events    
}
```

Reply:
Мы не можем возвращать ссылки на объекты с лайфтаймами, равными лайфтайму функции, так что, если думать в предложенном тобой направлении, то придётся возвращать не ссылку, а клон сета. Думаю, в целях уменьшения количества ненужных веток исполнения я так и сделаю пока, а потом можно подумать над лучшим решением